### PR TITLE
Guard against empty output_size argument for adaptive pool methods

### LINKED ISSUE #78868: `adapative_{avg, max}_pool` will crash in the backward pass on cuda on empty output size
### 🐛 Describe the bug

`adapative_{avg, max}_pool` will crash in the backward pass on cuda

```python
import torch

input = torch.rand([1, 64, 10, 9, 8], dtype=torch.float32, device='cuda', requires_grad=True)
output_size = 0

res = torch.nn.functional.adaptive_avg_pool3d(input, output_size)
print('forward')

res.sum().backward()
# Crash
print('backward')
```

It will try to compute the gradient based on an empty tensor.

### Versions

pytorch: 1.11.0
cuda: 11.3

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @ngimel

### PR DESCRIPTION
**Pull request by anshradh**
Fixes #78868

### CODE REVIEW

**Review by ngimel — COMMENTED** [review]
Please add the tests

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Test errors are real
